src/mono/wasm/Wasm.Build.Tests/TestAppScenarios/DownloadThenInitTests.cs

```diff
@@ -24,12 +24,20 @@ public DownloadThenInitTests(ITestOutputHelper output, SharedBuildPerTestClassFi
     [InlineData("Release")]
     public async Task NoResourcesReFetchedAfterDownloadFinished(string config)
     {
-        CopyTestAsset("WasmBasicTestApp", "DownloadThenInitTests", "App");
-        BuildProject(config);
-
-        var result = await RunSdkStyleAppForBuild(new(Configuration: config, TestScenario: "DownloadThenInit"));
+        ProjectInfo info = CopyTestAsset(config, aot: false, "WasmBasicTestApp", "DownloadThenInitTests", "App");
+        bool isPublish = false;
+        BuildTemplateProject(info,
+            new BuildProjectOptions(
+                info.Configuration,
+                info.ProjectName,
+                BinFrameworkDir: GetBinFrameworkDir(info.Configuration, isPublish),
```

Comment:
I don't think we should be passing this information. It should be computed from the parameters

Reply:
The same should be true for `ExpectedFileType`

Reply:
```
BuildTemplateProject(info,
    new BuildProjectOptions(
        info.Configuration,
        info.ProjectName,
        BinFrameworkDir: GetBinFrameworkDir(info.Configuration, isPublish),
        ExpectedFileType: GetExpectedFileType(info, isPublish: isPublish),
        IsPublish: isPublish
));
```
This is a boiler plate that we have in every test. We should find a way to simplify it. I believe we should end up with `BuildProject(new(info, configuration))` in most cases. If there are other parameters, like Aot, Relink, etc, just append those to the options `BuildProject(new(info, configuration, Aot: true))`

Reply:
`BuildOptions` has ~15 fields, I would prefer to avoid converting them to method args. I agree with the other ideas, I'll go towards making it:
```
private _defaultBuildOptions = new BuildOptions();
BuildProject(
        ProjectInfo info,
        string configuration,
        BuildOptions buildOptions = _defaultBuildOptions))
```

Reply:
BuildOptions as structure is fine! I'm just saying test author shouldn't be responsible for computing `GetBinFrameworkDir(info.Configuration, isPublish)` (none of the tests should be). Instead it should be computed by the combination of TestAsset+BuildArgs (aot, publish, config, etc)